libcxx/test/std/utilities/variant/variant.visit.member/robust_against_adl.pass.cpp

```diff
@@ -0,0 +1,48 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// UNSUPPORTED: c++03, c++11, c++14, c++17, c++20, c++23
+// The tested functionality needs deducing this.
+// UNSUPPORTED: clang-16 || clang-17 || apple-clang
```

Comment:
clang-16 and clang-17 do not work and we know clang-18 works. So we don't need to test these old versions. They never magically start to work. The current version of AppleClang does not work, a future version will start to work. We want to be notified when that happens event so we can add the proper AppleClang restrictions. When the test passes with AppleClang the test will fail since it was expected to fail. Then we can update the filter.

```suggestion
// UNSUPPORTED: clang-16 || clang-17
// XFAIL:  apple-clang
```